apps/desktop/src/features/spellcheck/spellcheckCore.ts

```diff
@@ -0,0 +1,71 @@
+// Types (mirrored from worker)
+export type SpellcheckWorker = Worker;
+
+class SpellcheckCore {
+  private worker: Worker;
+  private pendingRequests = new Map<string, (data: any) => void>();
+  private listeners = new Set<(event: any) => void>();
```

Comment:
Is there a better way to type this than `any`? Only do this if adding types won't make it too complicated.